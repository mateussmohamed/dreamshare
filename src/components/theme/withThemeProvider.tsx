import React, { Component, FC } from 'react'
import { ThemeProvider } from 'styled-components'

import theme from './index'

const getDisplayName = (component: FC) =>
  component.displayName || component.name || 'Component'

const withThemeProvider = (WrappedComponent: FC) =>
  class extends Component {
    static displayName = `withThemeProvider(${getDisplayName(
      WrappedComponent,
    )})`

    render() {
      return (
        <ThemeProvider theme={theme}>
          <WrappedComponent {...this.props} />
        </ThemeProvider>
      )
    }
  }

export default withThemeProvider
