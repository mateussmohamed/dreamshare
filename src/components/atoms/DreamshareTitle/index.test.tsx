import React from 'react'

import { DreamshareTitle } from './index'
import withThemeProvider from '../../theme/withThemeProvider'

describe('[<DreamshareTitle />]', () => {
  const wrap = props => withThemeProvider(() => <DreamshareTitle {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ children: 'I am a Title' })

    expect(wrapper).toMatchSnapshot()
  })
})
