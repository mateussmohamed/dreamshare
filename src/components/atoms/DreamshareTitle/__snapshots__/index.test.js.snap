// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<DreamshareTitle />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "DreamshareTitle-fg36s1-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";color:",
          [Function],
          ";font-weight:bold;text-align:center;@media only screen and (min-width:320px){font-size:26px;margin:35px 0 15px 0;}@media only screen and (min-width:768px){margin:95px 0 35px 0;}",
        ],
      },
      "displayName": "DreamshareTitle",
      "render": [Function],
      "styledComponentId": "DreamshareTitle-fg36s1-0",
      "target": "h1",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  I am a Title
</StyledComponent>
`;
